codec-http2/src/main/java/io/netty/handler/codec/http2/HpackEncoder.java

```diff
@@ -351,15 +352,15 @@ HpackHeaderField getHeaderField(int index) {
      * Returns the header entry with the lowest index value for the header field. Returns null if
      * header field is not in the dynamic table.
      */
-    private HeaderEntry getEntry(CharSequence name, CharSequence value) {
+    private HeaderEntry getEntryInsensitive(CharSequence name, CharSequence value) {
         if (length() == 0 || name == null || value == null) {
             return null;
         }
         int h = AsciiString.hashCode(name);
         int i = index(h);
         for (HeaderEntry e = headerFields[i]; e != null; e = e.next) {
             // To avoid short circuit behavior a bitwise operator is used instead of a boolean operator.
-            if (e.hash == h && (equalsConstantTime(name, e.name) & equalsConstantTime(value, e.value)) != 0) {
+            if (e.hash == h && equalsVariableTime(name, e.name) && equalsVariableTime(value, e.value)) {
```

Comment:
Do you think there would be a performance benefit from comparing the value before the name? In my minds eye, the value will typically have a larger range than the name (which we already know has the same hash so it's very likely to be the same as this entry). For example, we can skip iterating over all chars in the name for headers like:
```
content-length: 0
content-length: 1
```

Reply:
That's an interesting idea.  I don't want to go through the benchmarking again, but it passes the sniff test.  Done.

Reply:
@bryce-anderson that's a very nice idea!